packages/react-scripts/template/src/serviceWorker.js

```diff
@@ -66,11 +66,21 @@ function registerValidSW(swUrl) {
               // It's the perfect time to display a "New content is
               // available; please refresh." message in your web app.
               console.log('New content is available; please refresh.');
+
+              // Execute callback
+              if (config.onUpdate) {
```

Comment:
To keep register backwards compatible.
We should let user call serviceWorker.register(); without passing the config object (as per documentation).

So we should check config for undefined, at both places config.onUpdate and config.onSuccess 
or else user will get 
Uncaught TypeError: Cannot read property 'onUpdate' of undefined

May be it was just missed.

Reply:
This is released and causing the following error in 2.0: 
```
Uncaught TypeError: Cannot read property 'onSuccess' of undefined
    at ServiceWorker.n.onstatechange (serviceWorker.js:82)
```

Reply:
The check should be `if (config && config.onUpdate) {`

Reply:
Thanks, fixed in #5368.